arches/app/utils/data_management/resources/formats/htmlfile.py

```diff
@@ -0,0 +1,139 @@
+import os
+import logging
+from functools import lru_cache
+from uuid import UUID
+from io import StringIO
+from .format import Writer
+from django.template import Context, Template, TemplateDoesNotExist
+from django.template.loader import get_template
+from arches.app.models.models import GraphModel
+from arches.app.models.resource import Resource
+from arches.app.models.system_settings import settings
+from pathlib import Path
+
+
+logger = logging.getLogger(__name__)
+
+
+class HtmlWriter(Writer):
+    def __init__(self, **kwargs):
+        super(HtmlWriter, self).__init__(**kwargs)
+
+    @staticmethod
+    def get_graphids_with_export_template():
+        """
+        get a list of graphids that have html export templates available.
+        """
+        valid_graphs = []
+        existing_graphids = GraphModel.objects.filter(isresource=True).values_list("graphid", flat=True)
+        for existing_graphid in existing_graphids:
+            if HtmlWriter.has_html_template(existing_graphid):
+                valid_graphs.append(str(existing_graphid))
+
+        return valid_graphs
+
+    @staticmethod
+    @lru_cache(maxsize=None)
+    def has_html_template(graphid=None):
+        """
+        Check an html template for the graph is available
+        """
+        result = False
+        try:
+            temp = get_template(f"html_export/{graphid}.htm")
```

Comment:
Can you use os.path here rather than writing out the path? Also, it is possible that there is a template with the extension `.html` rather than `.htm`. It might be good to handle for that with something like: 
```
        for extension in ('htm', 'html'):
            try:
                path = os.path.join("html_export", str(graphid))
                get_template(f"{path}.{extension}")
                result = True
            except TemplateDoesNotExist as e:
                pass
```

Reply:
Agreed. I originally used html and then changed it to htm in order to match the file extension used for all the other templates.